libs/langchain/langchain/document_loaders/parsers/docai.py

```diff
@@ -0,0 +1,292 @@
+        else:
+            try:
+                from google.api_core.client_options import ClientOptions
+                from google.cloud.documentai import DocumentProcessorServiceClient
+            except ImportError:
+                raise ImportError(
+                    "documentai package not found, please install it with"
+                    " `pip install google-cloud-documentai`"
+                )
+            options = ClientOptions(
+                api_endpoint=f"{location}-documentai.googleapis.com"
+            )
+            self._client = DocumentProcessorServiceClient(client_options=options)
+
+    def lazy_parse(self, blob: Blob) -> Iterator[Document]:
+        """Parses a blob lazily.
+
+        Args:
+            blobs: a Blob to parse
+
+        This is a long-running operations! A recommended way is to batch
+            documents together and use `batch_parse` method.
+        """
+        yield from self.batch_parse([blob], gcs_output_path=self._gcs_output_path)
+
+    def batch_parse(
+        self,
+        blobs: Sequence[Blob],
+        gcs_output_path: Optional[str] = None,
+        timeout_min: int = 60,
+        check_in_interval_sec: int = 60,
+    ) -> Iterator[Document]:
+        """Parses a list of blobs lazily.
+
+        Args:
+            blobs: a list of blobs to parse
+            gcs_output_path: a path on GCS to store parsing results
+            timeout_min: a timeout to wait for DocAI to complete, in minutes
+            check_in_interval_sec: an interval (in sec) to wait until next check
+                whether parsing operations have been completed
+        This is a long-running operations! A recommended way is to decouple
+            parsing from creating Langchain Documents:
+            >>> operations = parser.docai_parse(blobs, gcs_path)
+            >>> parser.is_running(operations)
+            You can get operations names and save them:
+            >>> names = [op.operation.name for op in operations]
+            And when all operations are finished, you can use their results:
+            >>> operations = parser.operations_from_names(operation_names)
+            >>> results = parser.get_results(operations)
+            >>> docs = parser.parse_from_results(results)
+        """
+        output_path = gcs_output_path if gcs_output_path else self._gcs_output_path
+        if output_path is None:
+            raise ValueError("An output path on GCS should be provided!")
+        operations = self.docai_parse(blobs, gcs_output_path=output_path)
+        operation_names = [op.operation.name for op in operations]
+        logger.debug(
+            f"Started parsing with DocAI, submitted operations {operation_names}"
+        )
+        is_running, time_elapsed = True, 0
+        while is_running:
+            is_running = self.is_running(operations)
+            if not is_running:
+                break
+            time.sleep(check_in_interval_sec)
+            time_elapsed += 1
```

Comment:
is this supposed to be adding 1 or adding check_in_interval_sec?

Reply:
Good catch, fixed!